### PR TITLE
Add device role comparison table

### PR DESCRIPTION
**Pull request by tekstrand**
## What
1. Adds a comparison table of device role default settings

## Why
1. Misunderstanding of Device Roles: There's a common misunderstanding about the roles of devices within the mesh
2. There's a need for improved educational resources and documentation to clarify device roles and optimal configurations within the network.

# IMPORTANT
For feedback or discussion on specific values, please either comment in the [Google Sheet](https://docs.google.com/spreadsheets/d/1sN0EzJEhlG_shaCdwolba_jQLpuApypmQ3KpjguMYpU/edit#gid=72390815) OR comment inline under the `Files changed` tab.

## Screenshots
Screenshot 2024-02-04 at 8 50 17 PM

### CODE REVIEW

**Comment by pdxlocations** [comment]
I'll have a good look through this later, just mentioning this would negate #934 @charminULTRA

**Comment by pdxlocations** [comment]
I'm a fan of checkmark and x images a la: https://github.com/RicInNewMexico/Meshtastic-Antenna-Reports
Just saying.

**Comment by garthvh** [comment]
This information is in the comments in the protobuf comments and the apple apps, so needs to be updated in all three places.  I also think most of these are still too long. Nobody reads more than a sentence ever.

**Comment by charminULTRA** [comment]
> This information is in the comments in the protobuf comments and the apple apps, so needs to be updated in all three places. I also think most of these are still too long. Nobody reads more than a sentence ever.

Most of this PR's versions are too long, or the ones in the protobufs/apps are too long?

**Comment by charminULTRA** [comment]
I like the comparison table a lot.

I like that the Device Descriptions are in plain English rather than what we've had as very technical. My PR is a mixture of plain and technical language. The one I think is _too_ simple in this PR is Repeater, since it's a derivative of Router, and should still emphasize that it's intended to be placed in high coverage areas and isn't a magic wand that suddenly makes the node into what I'll call a "traditional" radio repeater, since power levels are still likely to be identical to any node.

File: docs/configuration/radio/device.mdx [hunk 1]
```diff
@@ -11,24 +11,34 @@ The device config options are: Role, Serial Output, and Debug Log. Device config
...
+CLIENT_MUTE   |Similar to CLIENT but does not forward packets.                                           |Situations where a node needs to participate in the network without assisting in packet routing, reducing network load.               
+ROUTER        |Devices placed for optimal routing, does not interact with client applications.           |Strategic placement in high-coverage areas to enhance network routing capabilities without the need for client interaction.           
+ROUTER_CLIENT |Acts as both a router and client device.                                                  |Devices that need to serve as both a communication point for users and a routing node to enhance network coverage.                    
+REPEATER      |Focuses on extending range with minimal network overhead.                                 |Extending the network's range by relaying messages without adding extra network overhead. Ideal for cleaner network topology.
```
Comment (tekstrand) [inline docs/configuration/radio/device.mdx:22]: @charminULTRA any thoughts on these?

`Focuses on extending range with minimal network overhead.` -> `Dedicated node that focuses solely on extending mesh network coverage by relaying messages, without appearing in the network's active node list or being directly addressable.`

`Extending the network's range by relaying messages without adding extra network overhead. Ideal for cleaner network topology.` -> `Best positioned in elevated or strategic locations to maximize the mesh network's overall coverage and connectivity.`

Comment (charminULTRA) [reply docs/configuration/radio/device.mdx:22]: @tekstrand I like those edits! Agreed.

Overall I like your versions better, as I'm generally on the side of making our Docs almost entirely in Plain English, but others may chime in that they want to keep some of the technical information, esp for Tracker and Sensor.

One small edit to Sensor I'll put in the Google Doc.

Comment (tekstrand) [reply docs/configuration/radio/device.mdx:22]: I am also expecting people to request keeping the technical documentation. If that happens, my suggestion is going to be putting the information in it's own dedicated table.

**Comment by tekstrand** [comment]
♻️

File: docs/configuration/radio/device.mdx [hunk 2]
```diff
@@ -11,24 +11,34 @@ The device config options are: Role, Serial Output, and Debug Log. Device config
...
+| CLIENT_HIDDEN  | Yes                     | Yes            | Lowest            | Known only | No                  | Yes                   |
+| TRACKER        | Optional                | No             | Low               | No         | No                  | Yes                   |
+| LOST_AND_FOUND | Optional                | No             | Regular           | No         | No                  | Yes                   |
+| SENSOR         | Optional                | No             | Lowest            | No         | No                  | Yes                   |
+| TAK            | Yes                     | Optional       | Low               | Yes        | No                  | Yes                   |
+| ROUTER         | No                      | No             | High              | Yes        | Yes                 | Yes                   |
+| ROUTER_CLIENT  | Yes                     | Yes            | Highest           | Yes        | Yes                 | Yes                   |
+| REPEATER       | No                      | No             | High              | Yes        | Yes                 | No                    |
```
Comment (pdxlocations) [inline docs/configuration/radio/device.mdx:41]: REPEATER has prioritized routing

File: docs/configuration/radio/device.mdx [hunk 3]
```diff
@@ -11,24 +11,34 @@ The device config options are: Role, Serial Output, and Debug Log. Device config
...
+| ROUTER_CLIENT  | Combination of both ROUTER and CLIENT.                                                                                   | Devices in a strategic position for priority routing that need to also serve as a standard CLIENT.                                     |                            
+
+#### Role Comparison
+| Device Role    | Bluetooth/Wi-Fi Enabled | Screen Enabled | Power Consumption | Retransmit | Prioritized Routing | Visible in Nodes List |
+| -------------- | ----------------------- | -------------- | ----------------- | ---------- | ------------------- | --------------------- |
+| CLIENT         | Yes                     | Yes            | Regular           | Yes        | No                  | Yes                   |
+| CLIENT_MUTE    | Yes                     | Yes            | Lowest            | No         | No                  | Yes                   |
+| CLIENT_HIDDEN  | Yes                     | Yes            | Lowest            | Known only | No                  | Yes                   |
```
Comment (pdxlocations) [inline docs/configuration/radio/device.mdx:34]: CLIENT_HIDDEN
- Retransmit is Local Only
- Visible in Nodes List, I haven't tested but I assume NO.  (config.device.node_info_broadcast_secs = UINT32_MAX)

File: docs/configuration/radio/device.mdx [hunk 4]
```diff
@@ -11,24 +11,34 @@ The device config options are: Role, Serial Output, and Debug Log. Device config
...
+| TRACKER        | Optional                | No             | Low               | No         | No                  | Yes                   |
+| LOST_AND_FOUND | Optional                | No             | Regular           | No         | No                  | Yes                   |
+| SENSOR         | Optional                | No             | Lowest            | No         | No                  | Yes                   |
```
Comment (pdxlocations) [inline docs/configuration/radio/device.mdx:37]: SENSOR would use more power as it broadcasts telemetry more often ? (unless is_power_saving = true ?)

File: docs/configuration/radio/device.mdx [hunk 5]
```diff
@@ -11,24 +11,34 @@ The device config options are: Role, Serial Output, and Debug Log. Device config
...
+| CLIENT_MUTE    | Same as CLIENT but does not forward packets from other devices.                                                          | Situations where a device needs to participate in the network without assisting in packet routing, reducing network load.              |
+| CLIENT_HIDDEN  | Only broadcasts as needed for stealth or power savings, still rebroadcasts known meshes.                                 | Use in clandestine operations or to reduce airtime/power consumption while still participating in the network.                         |
+| TRACKER        | Broadcasts GPS position packets as priority.                                                                             | Tracking the location of individuals or assets, especially in scenarios where timely and efficient location updates are critical.      |
+| LOST_AND_FOUND | Broadcasts location to default channel regularly for recovery efforts.                                                   | Used for recovery efforts of a lost device.                                                                                            |
```
Comment (pdxlocations) [inline docs/configuration/radio/device.mdx:22]: I'd suggest something like "broadcasts a location message" so we know it sends a text, not just a location packet

File: docs/configuration/radio/device.mdx [hunk 6]
```diff
@@ -11,24 +11,34 @@ The device config options are: Role, Serial Output, and Debug Log. Device config
...
+| SENSOR         | Broadcasts priority telemetry packets, optimized for sensor data collection.                                             | Deploying in scenarios where gathering environmental or other sensor data is crucial, with efficient power usage and frequent updates. |
```
Comment (pdxlocations) [inline docs/configuration/radio/device.mdx:23]: I don't understand "Broadcasts priority telemetry packets"

File: docs/configuration/radio/device.mdx [hunk 7]
```diff
@@ -11,24 +11,34 @@ The device config options are: Role, Serial Output, and Debug Log. Device config
...
+| TAK            | Optimized for ATAK system communication, reduces routine broadcasts.                                                     | Integration with ATAK systems for communication in tactical or coordinated operations.                                                 |
+| REPEATER       | Dedicated device that focuses solely on extending mesh network coverage by relaying messages. Not visible in Nodes list. | Best positioned in strategic locations to maximize the network's overall coverage not shown in topology.                               |
```
Comment (pdxlocations) [inline docs/configuration/radio/device.mdx:25]: maybe "... by relaying messages with minimal overhead" ?

Is "Best positioned in strategic locations to maximize the network's overall coverage not shown in topology." missing punctuation?

File: docs/configuration/radio/device.mdx [hunk 8]
```diff
@@ -11,24 +11,34 @@ The device config options are: Role, Serial Output, and Debug Log. Device config
...
+| ROUTER         | Dedicated device that focuses solely on extending mesh network coverage by relaying messages. Visible in Nodes list.     | Best positioned in strategic locations to maximize the network's overall coverage shown in topology.                                   |
```
Comment (pdxlocations) [inline docs/configuration/radio/device.mdx:26]: "Best positioned in strategic locations to maximize the network's overall coverage shown in topology." missing punctuation

File: docs/configuration/radio/device.mdx (hunk 1 above)
Comment (pdxlocations) [reply docs/configuration/radio/device.mdx]: > I am also expecting people to request keeping the technical documentation. If that happens, my suggestion is going to be putting the information in it's own dedicated table.

I think once the "plain english" gets tightened up, we could ask some devs to chime in on the technical side.  I think some tech info would be beneficial otherwise it's "take our word for it, you want this role"  For example "When used in conjunction with power.is_power_saving = true, nodes will wake up, send position, and then sleep for position.position_broadcast_secs seconds."  This needs to be mentioned somewhere or this feature wont be used.

File: docs/configuration/radio/device.mdx (hunk 2 above)
Comment (tekstrand) [reply docs/configuration/radio/device.mdx:41]: That's what this shows. Spreadsheet might be easier to read https://docs.google.com/spreadsheets/d/1sN0EzJEhlG_shaCdwolba_jQLpuApypmQ3KpjguMYpU/edit#gid=507821102

File: docs/configuration/radio/device.mdx (hunk 5 above)
Comment (tekstrand) [reply docs/configuration/radio/device.mdx:22]: "Broadcasts location as message to default channel regularly for recovery efforts."

File: docs/configuration/radio/device.mdx (hunk 4 above)
Comment (tekstrand) [reply docs/configuration/radio/device.mdx:37]: Setting as "High"

File: docs/configuration/radio/device.mdx (hunk 6 above)
Comment (tekstrand) [reply docs/configuration/radio/device.mdx:23]: Current docs state "Telemetry packets sent from this device will be higher priority"

**Comment by tekstrand** [comment]
♻️

**Review by charminULTRA — APPROVED** [review]
I'm good with the current version, great work.

File: docs/configuration/radio/device.mdx [hunk 9]
```diff
@@ -11,24 +11,33 @@ The device config options are: Role, Serial Output, and Debug Log. Device config
...
+
+#### Role Comparison
+| Device Role    | Bluetooth/Wi-Fi Enabled | Screen Enabled | Power Consumption | Retransmit | Prioritized Routing | Visible in Nodes List |
+| -------------- | ----------------------- | -------------- | ----------------- | ---------- | ------------------- | --------------------- |
+| CLIENT         | Yes                     | Yes            | Regular           | Yes        | No                  | Yes                   |
+| CLIENT_MUTE    | Yes                     | Yes            | Lowest            | No         | No                  | Yes                   |
+| CLIENT_HIDDEN  | Yes                     | Yes            | Lowest            | Local only | No                  | No                    |
+| TRACKER        | Optional                | No             | Regular           | No         | No                  | Yes                   |
+| LOST_AND_FOUND | Optional                | No             | Regular           | No         | No                  | Yes                   |
+| SENSOR         | Optional                | No             | High              | No         | No                  | Yes                   |
```
Comment (pdxlocations) [inline docs/configuration/radio/device.mdx:36]: Why Optional for these three?

File: docs/configuration/radio/device.mdx (hunk 4 above)
Comment (pdxlocations) [reply docs/configuration/radio/device.mdx]: This one is tricky since I believe it's designed to be low power but only when power saving is enabled.

File: docs/configuration/radio/device.mdx (hunk 6 above)
Comment (pdxlocations) [reply docs/configuration/radio/device.mdx]: maybe "Broadcasts high-priority telemetry packets" ?

File: docs/configuration/radio/device.mdx (hunk 4 above)
Comment (tekstrand) [reply docs/configuration/radio/device.mdx]: For the purposes of this table, we are describing the initial state of the configuration. I think either Regular/High is correct.

File: docs/configuration/radio/device.mdx (hunk 9 above)
Comment (tekstrand) [reply docs/configuration/radio/device.mdx:36]: Honestly can't remember why. I don't like it anymore. I think we just want whatever the default is, which I actually do not know what that is.

**Comment by garthvh** [comment]
Would really help if we could actually compare the changed text, images and replacing the entirety 

> > This information is in the comments in the protobuf comments and the apple apps, so needs to be updated in all three places. I also think most of these are still too long. Nobody reads more than a sentence ever.
> 
> @garthvh Most of this PR's versions are too long, or the ones in the protobufs/apps are too long?
> 
> Also, we may want the Protobuf comments to be more technical than the apps or docs. Losing technical info may not be a good idea in protobufs.

Both are too long, the comments in the code can have extra details after the descriptions if necessary.

**Comment by charminULTRA** [comment]
> > > This information is in the comments in the protobuf comments and the apple apps, so needs to be updated in all three places. I also think most of these are still too long. Nobody reads more than a sentence ever.
> > 
> > 
> > @garthvh Most of this PR's versions are too long, or the ones in the protobufs/apps are too long?
> > Also, we may want the Protobuf comments to be more technical than the apps or docs. Losing technical info may not be a good idea in protobufs.
> 
> Both are too long, the comments in the code can have extra details after the descriptions if necessary.

Are you asking for a specific action item based on your comment? I'm not really following.

I think this PR is a huge improvement over the existing Docs, and even if you think they're still too long, I think we should merge this PR and worry about shortening afterwards.

**Comment by garthvh** [comment]
Slightly modified versions of these are also translated strings in the iOS app and are the protobuf comments, so there needs to be quite a bit of release coordination, making them consistent would be great.

If the role comparison part can be separated we can merge that sooner,  but the main table is going to be a bigger lift. 

The latest strings on iOS are here https://github.com/meshtastic/Meshtastic-Apple/blob/f61e91a282706b2da4a298d7b3843d17d5d13a9f/en.lproj/Localizable.strings#L70 not sure how many of the languages have translated them.

**Comment by tekstrand** [comment]
I'll split these out and get a PR open for just the description updates. Once those are finalized here we can get PRs open to the other various repos before a merge to docs.

Do we think there is a potential win long term by separating User facing docs vs developer docs/protobufs? Having those coupled obviously presents challenges, but I am not sure what benefits there are given my short tenure on this project.

**Comment by tekstrand** [comment]
♻️ PR is now trimmed down to just device comparison. New PR incoming in Draft state for updating main description.

**Comment by tekstrand** [comment]
@pdxlocations @rcarteraz planning to merge this tonight unless anyone requests changes. That will let us move forward with #973 since @garthvh was kind enough to get the in app descriptions updated already.

File: docs/configuration/radio/device.mdx [hunk 10]
```diff
@@ -30,6 +30,24 @@ Acceptable values:
...
+| CLIENT_HIDDEN  | Yes                     | Yes            | Lowest            | Local only | No                  | No                    |
+| TRACKER        | Yes                     | No             | Regular           | No         | No                  | Yes                   |
+| LOST_AND_FOUND | Yes                     | No             | Regular           | No         | No                  | Yes                   |
+| SENSOR         | Yes                     | No             | High              | No         | No                  | Yes                   |
+| TAK            | Yes                     | Optional       | Low               | Yes        | No                  | Yes                   |
+| ROUTER         | No                      | No             | High              | Yes        | Yes                 | Yes                   |
+| ROUTER_CLIENT  | Yes                     | Yes            | Highest           | Yes        | Yes                 | Yes                   |
+| REPEATER       | Yes                     | No             | High              | Yes        | Yes                 | No                    |
```
[the remark this answers is not in the document]
Comment (tekstrand) [reply docs/configuration/radio/device.mdx:48]: It is Yes, just a little hard to read here in markdown. https://docs.google.com/spreadsheets/d/1sN0EzJEhlG_shaCdwolba_jQLpuApypmQ3KpjguMYpU/edit#gid=507821102

[the remark this answers is not in the document]
Comment (tekstrand) [reply docs/configuration/radio/device.mdx:48]: Screenshot 2024-02-06 at 8 35 37 PM

**Comment by garthvh** [comment]
Looks good, I have updated strings in the iOS app, I will ask for some translations
